functions/Find-DbaOrphanedFile.ps1

```diff
@@ -0,0 +1,231 @@
+	            Throw
+            }
+
+            # Set the method to locally
+            $method = 'local'
+        }
+        else
+        {
+            # Set the method to remote
+            $method = 'remote'
+        }
+
+		#Write-Output "Attempting to connect to SQL Server.."
+		
+		$sourceserver = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+
+        # Get all the database files
+        $databasefiles = Get-SqlFileStructure -server $sourceserver
+
+        # Check if there are any database files retrieved
+        if($databasefiles.count -ge 1)
+        {
+            # Create the orphaned files variable
+            $orphanedfiles = @() 
+
+            # Getthe default data and log directories from the instance
+            $SourceDirectory += $sourceserver.RootDirectory + "\DATA"
+
+            # Check if the default file path is set
+            if (($sourceserver.Settings.DefaultFile).Length -eq 0) 
+            {
+	            $SourceDirectory += $sourceserver.Information.MasterDBPath
+	        }
+            else
+            {
+                $SourceDirectory += $sourceserver.Settings.DefaultFile
+            }
+
+            # Check if the default log path is set
+	        if (($sourceserver.Settings.DefaultLog).Length -eq 0) {
+	            $SourceDirectory = $sourceserver.Information.MasterDBLogPath
+	        }
+            else
+            {
+                $SourceDirectory += $sourceserver.Settings.DefaultLog
+            }
+
+            # Create the array to hold the files on disk
+            $diskfiles = @()
+
+            # Loop through each of the directories and get all the data and log file related files
+            foreach($directory in $SourceDirectory)
+            {
+                # Cleanup directory
+                if($directory.EndsWith("\")) 
+                {
+                    $directory = $directory.TrimEnd("\")
+                }
+                
+                # Check the method and change to UNC admin paths
+                if($method -eq 'remote')
+                {
+                    # Set the directory to administrative UNC paths
+                    $directory = Join-AdminUnc -servername $servername -FilePath $directory
+
+                }
+                
+                # Check if the path exists
+                if(Test-Path $directory)
+                {
+                    # Get the files on disk
+                    $diskfiles += Get-ChildItem -Path "$directory\*.*" -Include *.mdf,*.ldf,*.ndf 
+                }
+                
+            }
+
+            # If there are files found
+            if($diskfiles.count -ge 1)
+            {
+                switch($method)
+                {
+                    'local' { $dbfiles = $databasefiles.Tables[0].filename }
+                    'remote' { $dbfiles = $databasefiles.Tables[0].filename | ? {$_ -match ":"} | % {$_ -replace ':','$'} | % {"\\$servername\" + $_} }

```

Comment:
Oh snap, brain interpreted it as full commands. Correct, we do not use aliases/shortcuts, and in addition, we also cap. So this would be Where-Object and ForEach-Object